com.unity.shadergraph/Editor/Generation/Targets/CustomRenderTexture/CustomTextureNodes.cs

```diff
@@ -0,0 +1,176 @@
+
+        public const int OutputSlotSelf2DId = 5;
+        public const int OutputSlotSelfCubeId = 6;
+        public const int OutputSlotSelf3DId = 7;
+
+        public CustomTextureSelf()
+        {
+            name = "Custom Render Texture Self";
+            UpdateNodeAfterDeserialization();
+        }
+
+        protected int[] validSlots => new[] { OutputSlotSelf2DId, OutputSlotSelfCubeId, OutputSlotSelf3DId };
+
+        public sealed override void UpdateNodeAfterDeserialization()
+        {
+            AddSlot(new Texture2DMaterialSlot(OutputSlotSelf2DId, kOutputSlotSelf2DName, kOutputSlotSelf2DName, SlotType.Output, ShaderStageCapability.Fragment, false) { bareResource = true });
+            AddSlot(new CubemapMaterialSlot(OutputSlotSelfCubeId, kOutputSlotSelfCubeName, kOutputSlotSelfCubeName, SlotType.Output, ShaderStageCapability.Fragment, false) { bareResource = true });
+            AddSlot(new Texture3DMaterialSlot(OutputSlotSelf3DId, kOutputSlotSelf3DName, kOutputSlotSelf3DName, SlotType.Output, ShaderStageCapability.Fragment, false) { bareResource = true });
+            RemoveSlotsNameNotMatching(validSlots);
+        }
+
+        public override string GetVariableNameForSlot(int slotId)
+        {
+            switch (slotId)
+            {
+                case OutputSlotSelf2DId:
+                    return "UnityBuildTexture2DStructNoScale(_SelfTexture2D)";
+                case OutputSlotSelfCubeId:
+                    return "UnityBuildTextureCubeStruct(_SelfTextureCube)";
+                default:
+                    return "UnityBuildTexture3DStruct(_SelfTexture3D)";
+            }
+        }
+
+        public void GenerateNodeFunction(FunctionRegistry registry, GenerationMode generationMode)
+        {
+            // For preview only we declare CRT defines
+            if (generationMode == GenerationMode.Preview)
+            {
+                registry.builder.AppendLine("#if !defined(UNITY_CRT_PREVIEW_TEXTURE) && !defined(UNITY_CUSTOM_TEXTURE_INCLUDED)");
+                registry.builder.AppendLine("#define UNITY_CRT_PREVIEW_TEXTURE");
+                registry.builder.AppendLine("TEXTURE2D(_SelfTexture2D);");
+                registry.builder.AppendLine("SAMPLER(sampler_SelfTexture2D);");
+                registry.builder.AppendLine("float4 _SelfTexture2D_TexelSize;");
+                registry.builder.AppendLine("TEXTURECUBE(_SelfTextureCube);");
+                registry.builder.AppendLine("SAMPLER(sampler_SelfTextureCube);");
+                registry.builder.AppendLine("float4 _SelfTextureCube_TexelSize;");
+                registry.builder.AppendLine("TEXTURE3D(_SelfTexture3D);");
+                registry.builder.AppendLine("SAMPLER(sampler_SelfTexture3D);");
+                registry.builder.AppendLine("float4 sampler_SelfTexture3D_TexelSize;");
+                registry.builder.AppendLine("#endif");
+            }
+        }
+    }
+}
```

Comment:
Nit: I don't think we have a precedence of putting multiple shader definitions in the same file.

Seems like these should/could all be one big accessor node? @bencloward what do you think here?

Reply:
For time sensitivity RE: SRP2Core, you could make a JIRA and comment it here in case this change needs to be deferred.

Reply:
I would prefer to have these in three different files where the name of the file matches the name of the node - ie SliceIndexNode.cs, SizeNode.cs, etc. I don't mind having three different nodes, but nodes are always organized with one node per file.

Reply:
I moved the nodes to 3 different files with matching names :)